File: pset/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void){
    float dollars;

    do{
        dollars = get_float("Change owed (in dollars): ");
    }
    while(dollars < 0);
    int x = round(dollars * 100);
    int n = 0; // Number of coins
    int y = x; // Change owed

    while(y >= 25){
        y = y - 25;
        n++;
    }

    while(y >= 10 && y < 25){
        y = y - 10;
        n++;
    }

    while(y >= 5 && y < 10){
        y = y - 5;
        n++;
    }

    while(y >= 1 && y < 5){
        y = y - 1;
        n++;
    }
    printf("%i\n", n);
}

File: pset/pset6/cash/cash.py
from cs50 import get_float

while True:
	dollars = get_float("Change owed (in dollars): ")
	if dollars >= 0: break

x = round(dollars * 100) # Convert dollars to cents
n = 0 # Number of coins
y = x # Change owed

while y >= 25:
	y -= 25
	n += 1

while y >= 10 and y < 25:
	y -= 10
	n += 1

while y >= 5 and y < 10:
	y -= 5
	n += 1

while y >= 1 and y < 5:
	y -= 1
	n += 1

print(n)